pysindy/optimizers/constrained_sr3.py

```diff
@@ -271,20 +269,24 @@ def _update_full_coef_constraints(self, H, x_transpose_y, coef_sparse):
         rhs = rhs.reshape(g.shape)
         return inv1.dot(rhs)
 
+    def _calculate_penalty(self, xi: cp.Variable) -> cp.Expression:
+        thresholder = self.thresholder.lower()
+        if thresholder == "l1":
+            return self.threshold * cp.norm1(xi)
+        elif thresholder == "weighted_l1":
+            return cp.norm1(cp.multiply(np.ravel(self.thresholds), xi))
+        elif thresholder == "l2":
+            return self.threshold * cp.norm2(xi) ** 2
+        elif thresholder == "weighted_l2":
+            return cp.norm2(cp.multiply(np.ravel(self.thresholds), xi)) ** 2
+
```

Comment:
Let's make this a `@staticmethod` so it can be tested without constructing a whole `ConstrainedSR3` object.